File: app/src/main/java/com/github/kazy1991/remee/module/RemeeModule.java
package com.github.kazy1991.remee.module;

import android.content.Context;

import com.github.kazy1991.dependencykit.Module;
import com.github.kazy1991.remee.lazy.TwitterPackLazy;
import com.github.kazy1991.twitterpack.TwitterPack;


public class RemeeModule extends Module {
    private final Context context;

    public RemeeModule(Context context) {
        this.context = context;
        configure();
    }

    @Override
    protected void configure() {
        bind(TwitterPack.class).to(new TwitterPackLazy(context));
    }
}

File: app/src/main/java/com/github/kazy1991/remee/module/RemeeModule.kt
package com.github.kazy1991.remee.module

import android.content.Context

import com.github.kazy1991.dependencykit.Module
import com.github.kazy1991.remee.lazy.TwitterPackLazy
import com.github.kazy1991.twitterpack.TwitterPack


class RemeeModule(private val context: Context) : Module() {

    init {
        configure()
    }

    override fun configure() {
        bind(TwitterPack::class.java).to(TwitterPackLazy(context))
    }
}
